base/sim/FairMCApplication.cxx

```diff
@@ -512,6 +513,7 @@ void FairMCApplication::InitOnWorker()
     // and create a new sink on worker
     // moved to CloneForWorker and specific sink->CloneSink();
 
+    mtx.lock();
```

Comment:
I am not so happy, that you lock a mutex in `InitOnWorker` and unlock it in `InitGeometry`. What if things break in between and `InitGeometry` is never called?

To be honest, I don't understand, what you're protecting. If each thread has its own TFile, then there shouldn't be any races?

Anyway: Could you limit the area that needs locking to a very located area maybe?